crypten/mpc/primitives/arithmetic.py

```diff
@@ -327,21 +327,32 @@ def _arithmetic_function(self, y, op, inplace=False, *args, **kwargs):  # noqa:C
             else:  # ['mul', 'matmul', 'convNd', 'conv_transposeNd']
                 # NOTE: 'mul_' calls 'mul' here
                 # Must copy share.data here to support 'mul_' being inplace
-                result.share.set_(
-                    getattr(beaver, op)(result, y, *args, **kwargs).share.data
-                )
+                if comm.get().get_world_size() == 3:
+                    result.share.set_(
+                        getattr(resharing, op)(result, y, *args, **kwargs).share.data
+                    )
+                else:
+                    result.share.set_(
+                        getattr(beaver, op)(result, y, *args, **kwargs).share.data
+                    )
         else:
             raise TypeError("Cannot %s %s with %s" % (op, type(y), type(self)))
 
         # Scale by encoder scale if necessary
         if not additive_func:
             if public:  # scale by self.encoder.scale
                 if self.encoder.scale > 1:
+                    if comm.get().get_world_size() == 3:
```

Comment:
The if statement here is only a placeholder. Same for all the if statements elsewhere. We probably need to discuss how exactly we would like to fit RSS into CrypTen. Once we settle down on a plan, I would change the logic correspondingly.

Reply:
Rather than automatically using RSS when world_size > 2 and Beaver otherwise, I think this should be a configurable option since there are privacy-efficiency tradeoffs to using Beaver vs. RSS (e.g. 2-of-3 secret sharing).

It would be good to use something like:
```
# with multiplication_protocol in [beaver, resharing]:
result.share.set_(getattr(multiplication_protocol, op)(...))
```

Then within `resharing.py`, we can raise an exception if `world_size` is not 3.

I'm currently working on some code that will allow us to pass in a config file. We can supply a `multiplication_protocol` within that file to determine the method to use. For now, we can just use a class variable for this and set via a getter / setter.

Reply:
Thanks. That's a good idea. Will make the corresponding adjustments.